pkg/config/expression.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Reference is data struct that represents a reference to a variable.
+// Neither checks are performed, nor context is captured, just a structural
+// representation of a reference text
+type Reference struct {
+	GlobalVar bool
+	Group     string // should be empty if GlobalVar, otherwise optional
+	Module    string // should be empty if GlobalVar. otherwise required
+	Name      string // required
+}
+
+// SimpleVarToReference takes a string `$(...)` and transforms it to `Reference`
+func SimpleVarToReference(s string) (Reference, error) {
+	if !isSimpleVariable(s) {
+		return Reference{}, fmt.Errorf(
```

Comment:
is this retaining the behavior of reporting all (not yet supported) blueprint variables that are attempting string interpolation? or is it just reporting generically that we do not yet support string interpolation?

I think this is a case where we want to retain the existing behavior: all exact strings should be printed.

Reply:
> we want to retain the existing behavior: all exact strings should be printed.

There is no such behavior currently. Settings traversal will stop as soon it finds first offender and reports only it.

As for string-interpolation error it will still output the whole (but not all) offending string. The only change is removal of "hint" from error message, as it doesn't provide any value and adds complexity.